### PR TITLE
chore: fixed spelling issues

### PR DESCRIPTION
**Pull request by LeonFLK**
## fixes [KILTProtocol/ticket#312](https://github.com/KILTprotocol/ticket/issues/412)

Just quick fix of some annoying spelling issues in our sdk code, checked with Code Spell Checker for vs code.

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: src/delegation/DelegationNode.utils.ts [hunk 1]
```diff
@@ -20,10 +20,10 @@ import { IDelegationNode } from '../types/Delegation'
...
   delegation: IDelegationNode
 ): Uint8Array {
-  const permisssionsAsBitset: number = delegation.permissions.reduce(
+  const permissionsBitset: number = delegation.permissions.reduce(
     (accumulator, currentValue) => accumulator + currentValue
   )
   const uint8: Uint8Array = new Uint8Array(4)
-  uint8[0] = permisssionsAsBitset
+  uint8[0] = permissionsBitset
   return uint8
```
Comment (flechtnergalani) [inline src/delegation/DelegationNode.utils.ts]: this actually makes it less clear

File: src/did/Did.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export interface IDidDocumentPublicKey {
   publicKeyHex: string
 }
 
-export interface IDidDocumentPpties {
+export interface IDidDocumentProperties {
```
Comment (flechtnergalani) [inline src/did/Did.ts]: we export this, so I wouldn't want to change the name

File: src/did/Did.ts [hunk 3]
```diff
@@ -80,7 +80,7 @@ export interface IDidService {
 
 export interface IDidDocument
   extends IDidDocumentCore,
-    Partial<IDidDocumentPpties> {}
+    Partial<IDidDocumentProperties> {}
```
Comment (flechtnergalani) [inline src/did/Did.ts]: same

File: src/errorhandling/ExtrinsicError.ts [hunk 4]
```diff
@@ -50,7 +50,7 @@ export const ERROR_CTYPE_ALREADY_EXISTS: ExtrinsicError = new ExtrinsicError(
   ErrorCode.ERROR_CTYPE_ALREADY_EXISTS,
   'CTYPE already exists'
 )
-export const ERROR_ALEADY_ATTESTED: ExtrinsicError = new ExtrinsicError(
+export const ERROR_ALREADY_ATTESTED: ExtrinsicError = new ExtrinsicError(
```
Comment (flechtnergalani) [inline src/errorhandling/ExtrinsicError.ts]: Ouch :D

File: src/__integrationtests__/Attestation.spec.ts [hunk 5]
```diff
@@ -47,7 +47,7 @@ describe('When there is an attester, claimer and ctype drivers license', async (
   }, 60_000)
 
   it('should be possible to make a claim', () => {
-    const content = { name: 'Ralfi', age: 12 }
+    const content = { name: 'Ralph', age: 12 }
```
Comment (flechtnergalani) [inline src/__integrationtests__/Attestation.spec.ts]:
```suggestion
    const content = { name: 'Raph', age: 12 }
```

File: src/delegation/DelegationNode.utils.ts (hunk 1 above)
Comment (LeonFLK) [reply src/delegation/DelegationNode.utils.ts]: Yeah this might be named different

File: src/did/Did.ts (hunk 2 above)
Comment (LeonFLK) [reply src/did/Did.ts]: dependencies can be fixed quick, therefore I think it is worth the compatibility patching and having the name be in line with our naming convention.

File: src/did/Did.ts (hunk 3 above)
Comment (LeonFLK) [reply src/did/Did.ts]: just a consumer of IDidDocumentProperties

File: src/__integrationtests__/Attestation.spec.ts (hunk 5 above)
Comment (LeonFLK) [reply src/__integrationtests__/Attestation.spec.ts]: Yes ofc

File: src/did/Did.ts (hunk 2 above)
Comment (LeonFLK) [reply src/did/Did.ts]: In our github we only have one consumer of this interface

File: src/__integrationtests__/Attestation.spec.ts (hunk 5 above)
Comment (Dudleyneedham) [reply src/__integrationtests__/Attestation.spec.ts:50]: Agreed

File: src/__integrationtests__/Ctypes.spec.ts [hunk 6]
```diff
@@ -59,7 +59,7 @@ describe('When there is an CtypeCreator and a verifier', async () => {
       } as ICType['schema'],
     } as ICType)
 
-    const iAmNotThereWowner = CType.fromCType({
+    const iAmNotThereOwner = CType.fromCType({
```
Comment (Dudleyneedham) [inline src/__integrationtests__/Ctypes.spec.ts:62]: What do you mean? 

I am not there owner or I am not the owner?

File: src/__integrationtests__/Ctypes.spec.ts [hunk 7]
```diff
@@ -75,7 +75,7 @@ describe('When there is an CtypeCreator and a verifier', async () => {
       expect(iAmNotThere.verifyStored()).resolves.toBeFalsy(),
       expect(getOwner(iAmNotThere.hash)).resolves.toBeNull(),
       expect(getOwner('0x012012012')).resolves.toBeNull(),
-      expect(iAmNotThereWowner.verifyStored()).resolves.toBeFalsy(),
+      expect(iAmNotThereOwner.verifyStored()).resolves.toBeFalsy(),
```
Comment (Dudleyneedham) [inline src/__integrationtests__/Ctypes.spec.ts:78]: See above

File: src/attestation/Attestation.utils.ts [hunk 8]
```diff
@@ -39,15 +39,15 @@ export function compress(attestation: IAttestation): CompressedAttestation {
...
+ * @param attestation A compressed [[Attestation]] array that is reverted back into an object.
  *
  * @returns An object that has the same properties as an [[Attestation]].
  */
 
 export function decompress(attestation: CompressedAttestation): IAttestation {
   if (!Array.isArray(attestation) || attestation.length !== 5) {
     throw new Error(
-      'Compressed Attestation isnt an Array or has all the required data types'
+      "Compressed Attestation isn't an Array or has all the required data types"
```
Comment (Dudleyneedham) [inline src/attestation/Attestation.utils.ts:50]: Thank you 🍡

File: src/attestation/Attestation.utils.ts [hunk 9]
```diff
@@ -39,15 +39,15 @@ export function compress(attestation: IAttestation): CompressedAttestation {
 /**
  *  Decompresses an [[Attestation]] from storage and/or message into an object.
  *
- * @param attestation A compressesd [[Attestation]] array that is reverted back into an object.
+ * @param attestation A compressed [[Attestation]] array that is reverted back into an object.
```
Comment (Dudleyneedham) [inline src/attestation/Attestation.utils.ts:42]: Thank you

File: src/__integrationtests__/Ctypes.spec.ts (hunk 6 above)
Comment (LeonFLK) [reply src/__integrationtests__/Ctypes.spec.ts:62]: the iAmNotThere CType but with Owner

will probably rename it to iAmNotThereWithOwner

the Wowner prolly ment to abbreviate WithOwner

Comment (flechtnergalani) [reply src/__integrationtests__/Ctypes.spec.ts]: Jep!

File: src/did/Did.ts (hunk 2 above)
Comment (flechtnergalani) [reply src/did/Did.ts:68]: fine with it if you attach a PR that fixes the consumer

Comment (LeonFLK) [reply src/did/Did.ts:68]: This PR fixes the only consumer.

**Comment by LeonFLK** [comment]
The consumer is fixed in this PR

File: src/delegation/DelegationNode.utils.ts (hunk 1 above)
Comment (weichweich) [reply src/delegation/DelegationNode.utils.ts]: maybe rename this to `permissionFlags`?

**Pull request merged**